integration/simple-prototype-defaults/simple-prototype-defaults-test.ts

```diff
@@ -0,0 +1,24 @@
+import { Simple } from './simple';
+
+describe('simple', () => {
+  it('generates types correctly', () => {
+    const decodedUndefined = Simple.decode(Simple.encode(Simple.fromPartial({})).finish());
+
+    expect(Object.keys(decodedUndefined).includes('age')).toBeFalsy();
+    expect(decodedUndefined.hasOwnProperty('age')).toBeFalsy();
+
+    const decodedWithZero = Simple.decode(Simple.encode(Simple.fromPartial({
+      age: 0
+    })).finish());
+
+    expect(Object.keys(decodedWithZero).includes('age')).toBeFalsy();
+    expect(decodedWithZero.hasOwnProperty('age')).toBeFalsy();
```

Comment:
Ah right right, I'd expected this to be truthy, but it's the whole proto3 thing that optionals are not encoded, so presence is broken. But this will work for proto2, which is what the descriptors comein as.